src/fides/api/oauth/utils.py

```diff
@@ -339,6 +339,14 @@ def has_permissions(
     has_role: bool = _has_scope_via_role(
         token_data=token_data, client=client, endpoint_scopes=endpoint_scopes
     )
+
+    if not has_direct_scope and not has_role:
```

Comment:
Nit: I prefer not duplicating the permission check for the log; it's the kind of pattern that invites subtle errors later on. Try this:
```suggestion
    has_permissions = has_direct_scope or has_role
    if not has_permissions:
```